core/src/test/java/org/mule/tck/MuleTestUtils.java

```diff
@@ -596,17 +596,23 @@ public static Flow getTestFlow(String name, MuleContext context) throws Exceptio
     }
 
     @Deprecated
-    public static Service getTestService(String name,
+    public static synchronized Service getTestService(String name,
                                          Class<?> clazz,
                                          Map props,
                                          MuleContext context,
                                          boolean initialize) throws Exception
     {
+        Service service = context.getRegistry().get(name);
```

Comment:
I think it was (tests remain green), and even if it wasn't the whole purpose of that service is to be the test event's flowConstruct so this would be actually be an improvement if the service can be reused.